### PR TITLE
[out of core]Replaced typedef PairwiseMatches with abstract class Pai…

### PR DESCRIPTION
**Pull request by bjornpiltz**
…rwiseMatchesContainer.

### CODE REVIEW

File: src/openMVG/matching/indMatch.hpp [hunk 1]
```diff
@@ -69,10 +69,26 @@ inline std::istream& operator>>(std::istream & in, IndMatch & obj) {
...
-/// The structure used to store corresponding point indexes per images pairs
-typedef std::map< Pair, IndMatches > PairWiseMatches;
+/// A structure used to store corresponding point indexes per images pairs
+struct PairWiseMatches : public PairWiseMatchesContainer, public std::map< Pair, IndMatches >
+{
+  void insert(std::pair<Pair, IndMatches> && pairWiseMatches)override
+  {
+      std::map< Pair, IndMatches >::insert(std::forward<std::pair<Pair, IndMatches>>(pairWiseMatches));
+  }
+};
```
Comment (pmoulon) [inline src/openMVG/matching/indMatch.hpp]: You must add a serialize function for Cereal (something like this...):
```
// Serialization
  template <class Archive>
  void serialize( Archive & ar )  {
    ar(std::static_cast<std::map< Pair, IndMatches >>(*this));
  }
```

File: src/openMVG/matching/indMatch.hpp [hunk 2]
```diff
@@ -85,4 +107,9 @@ inline Pair_Set getPairs(const PairWiseMatches & matches)
 }  // namespace matching
 }  // namespace openMVG
 
+namespace cereal
+{
+    // This struct specialization will tell cereal which is the right way to serialize PairWiseMatches
```
Comment (pmoulon) [inline src/openMVG/matching/indMatch.hpp]: Does void serialize( Archive & ar )  { is still used since you have added this new specialization?

**Comment by bjornpiltz** [comment]
Yes `member_serialize` tells cereal to use `serialize` as opposed to
`load`/`save`.

**Comment by pmoulon** [comment]
For some other class I use serialize and don't need to add the specialization.
Does it was required for a given compiler, or does you just add it by security?

**Comment by bjornpiltz** [comment]
I ran into [this compile error](https://travis-ci.org/openMVG/openMVG/jobs/179709779#L970) since load/save is defined for std::map and serialise for PairWiseMatches.

**Comment by pmoulon** [comment]
Congrats your first PR was successfully merged:
https://github.com/openMVG/openMVG/commit/7d26f579beb27f3046e0c2e87ef021c3031bf7f8

Sorry for the delay.

**Pull request closed without merging**